src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -1280,6 +1284,11 @@ void install_signal_handlers() {
   PPC64_ONLY(set_signal_handler(SIGTRAP);)
   set_signal_handler(SIGXFSZ);
 
+  // This is just for initialization phase. os::initialize_jdk_signal_support() will overwrite
+  // the signal handler of BREAK_SIGNAL later. Intercepting the signal here reduces the risk
+  // that an attach client accidentally forces HotSpot to quit prematurely.
+  set_signal_handler(BREAK_SIGNAL);
+
```

Comment:
I would do this only if ReduceSignalHandling is inactive, since in that case we must not overwrite a SIGQUIT handler the app may have installed before us. We should have parsed arguments already at this point, no?

Reply:
Arguments have been parsed right here(os::init_2()), so ReduceSignalUsage is evaluated. 
Previous revision actually covers java with -Xrs. Attach won't send SIGQUIT until the SIGQUIT bit has been set. 
if we install JVM_HANDLE_XXX_SIGNAL conditionally, it will leave java processes with -Xrs unprotected.

Reply:
> if we install JVM_HANDLE_XXX_SIGNAL conditionally, it will leave java processes with -Xrs unprotected.

That's the idea. ReduceSignalUsage (Xrs) means that the hotspot should leave this signal and some others alone:

```
1. On Solaris and Linux, the signal masks for SIGINT, SIGTERM, SIGHUP,
   and SIGQUIT are not changed by the JVM, and signal handlers for
   these signals are not installed.
```
because, presumably, the user already installed a handler for this signal and does not want the JVM to change it. Matters if  the JVM is embedded into a foreign launcher which does its own signal handling.